agent/config/runtime_test.go

```diff
@@ -5640,6 +5644,37 @@ func TestLoad_IntegrationWithFlags(t *testing.T) {
 			rt.TLS.InternalRPC.VerifyOutgoing = true
 		},
 	})
+	run(t, testCase{
+		desc: "tls.defaults.verify_server_hostname implies tls.internal_rpc.verify_outgoing",
+		args: []string{
+			`-data-dir=` + dataDir,
+		},
+		json: []string{`
+			{
+				"tls": {
+					"defaults": {
+						"verify_server_hostname": true
+					}
+				}
+			}
+		`},
+		hcl: []string{`
+			tls {
+				defaults {
+					verify_server_hostname = true
+				}
+			}
+		`},
+		expected: func(rt *RuntimeConfig) {
+			rt.DataDir = dataDir
+
+			rt.TLS.Domain = "consul."
+			rt.TLS.NodeName = "thehostname"
+
+			rt.TLS.InternalRPC.VerifyServerHostname = true
+			rt.TLS.InternalRPC.VerifyOutgoing = true
+		},
+	})
```

Comment:
Shall we add a test case testing that `tls.internal_rpc.verify_outgoing` overrides `tls.defaults.verify_server_hostname`?

Reply:
@fulviodenza , I am not sure if this test case has been added. Could you please confirm this? Thanks.

Reply:
yes, sorry, didn't push a change. Now should have been added